src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1071,4 +1072,152 @@ function ($a) {
+                                                    ->defaultValue(array())
+                                                    // Deal with XML config
+                                                    ->beforeNormalization()
+                                                        ->always()
+                                                        ->then(function ($v) {
+                                                            return $this->fixXmlArguments($v);
+                                                        })
+                                                    ->end()
+                                                    ->validate()
+                                                        ->ifTrue(function ($v) {
+                                                            return !is_array($v);
+                                                        })
+                                                        ->thenInvalid('Arguments must be an array (got %s).')
+                                                    ->end()
+                                                ->end()
+                                                ->enumNode('retry_strategy')
+                                                    ->values(array(null, 'constant', 'exponential'))
+                                                    ->defaultNull()
+                                                ->end()
+                                                ->variableNode('retry_strategy_options')
+                                                    ->validate()
+                                                        ->ifTrue(function ($v) {
+                                                            return !is_array($v);
+                                                        })
+                                                        ->thenInvalid('Arguments must be an array (got %s).')
+                                                    ->end()
+                                                ->end()
+                                                ->arrayNode('thresholds')
+                                                    ->addDefaultsIfNotSet()
+                                                    ->children()
+                                                        ->integerNode('warning')->defaultNull()->end()
+                                                        ->integerNode('critical')->defaultNull()->end()
+                                                    ->end()
+                                                ->end()
+                                            ->end()
+                                            ->validate()
+                                                ->ifTrue(function ($config) {
+                                                    return 'constant' === $config['retry_strategy'] && !array_key_exists('max', $config['retry_strategy_options']);
```

Comment:
Instead of these, can't we explicitly set the retry strategy options in `retry_strategy.constant` and `retry_strategy.exponential`? The only thing to validate after is that we only have one retry strategy.